### PR TITLE
Draft rest 1.1 doc for review

### PR DESCRIPTION
**Pull request by yao-msft**
I directly modified on the v1.0 spec for easier review of what changed between 1.1.0 and 1.0.0.

Will send another pr with copy of v1.1.0 schema after finalized.

### CODE REVIEW

File: documentation/WinGet-1.0.0.yaml [hunk 1]
```diff
@@ -678,6 +699,43 @@ components:
...
+        Agreement:
+          description: The agreement text content.
+          nullable: true
+          type: string
+          minLength: 3
+          maxLength: 10000
+        AgreementUrl:
+          description: The agreement url.
+          $ref: '#/components/schemas/Url'
+      required:
```
Comment (JohnMcPMS) [inline documentation/WinGet-1.0.0.yaml:720]: Is there no "one of" concept in the schema definition?  Or should we have two subtypes and an option?

File: documentation/WinGet-1.0.0.yaml [hunk 2]
```diff
@@ -535,12 +547,16 @@ paths:
 
   # Server API Calls
   /information:
+    parameters:
+      - $ref: '#/components/parameters/APIVersion'
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:551]: >- $ref: '#/components/parameters/APIVersion' [](http://example.com/codeflow?start=6&length=44)

remove

File: documentation/WinGet-1.0.0.yaml [hunk 3]
```diff
@@ -578,6 +594,11 @@ components:
       type: string
       pattern: "^([0-9]+\\.){0,3}(\\*|[0-9]+)$"
       maxLength: 128
+      
+    Windows-Package-Manager:
+      description: Custom header value to be passed along all rest calls from winget.
+      type: string
+      maxLength: 10000
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:601]: >10000 [](http://example.com/codeflow?start=17&length=5)

1024

File: documentation/WinGet-1.0.0.yaml [hunk 4]
```diff
@@ -678,6 +699,43 @@ components:
       items:
         $ref: '#/components/schemas/Tag'
 
+    Agreement:
+      description: One Agreement entry.
+      type: object
+      properties:
+        AgreementLabel:
+          description: The label of the agreement. i.e. EULA, AgeRating, etc
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:707]: >The label of the agreement. i.e. EULA, AgeRating, etc [](http://example.com/codeflow?start=23&length=53)

needs to be localized

File: documentation/WinGet-1.0.0.yaml [hunk 5]
```diff
@@ -678,6 +699,43 @@ components:
...
+          type: string
+          minLength: 1
+          maxLength: 40
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:710]: >40 [](http://example.com/codeflow?start=21&length=2)

100

File: documentation/WinGet-1.0.0.yaml [hunk 6]
```diff
@@ -678,6 +699,43 @@ components:
...
+        Agreement:
+          description: The agreement text content.
+          nullable: true
+          type: string
+          minLength: 3
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:715]: >3 [](http://example.com/codeflow?start=21&length=1)

1

File: documentation/WinGet-1.0.0.yaml [hunk 7]
```diff
@@ -678,6 +699,43 @@ components:
...
+      description: One Agreement entry.
+      type: object
+      properties:
+        AgreementLabel:
+          description: The label of the agreement. i.e. EULA, AgeRating, etc
+          type: string
+          minLength: 1
+          maxLength: 40
+        Agreement:
+          description: The agreement text content.
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:712]: >The agreement text content. [](http://example.com/codeflow?start=23&length=27)

either agreement or url is required

File: documentation/WinGet-1.0.0.yaml [hunk 8]
```diff
@@ -678,6 +699,43 @@ components:
...
+          nullable: true
+          type: string
+          minLength: 3
+          maxLength: 10000
+        AgreementUrl:
+          description: The agreement url.
+          $ref: '#/components/schemas/Url'
+      required:
+        - AgreementLabel
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:721]: remove

File: documentation/WinGet-1.0.0.yaml [hunk 9]
```diff
@@ -678,6 +699,43 @@ components:
...
+
+    Agreements:
+      description: List of agreements.
+      type: array
+      nullable: true
+      uniqueItems: true
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:727]: >uniqueItems: true [](http://example.com/codeflow?start=6&length=17)

remove

File: documentation/WinGet-1.0.0.yaml [hunk 10]
```diff
@@ -678,6 +699,43 @@ components:
...
+      maxItems: 128
+      items:
+        $ref: '#/components/schemas/Agreement'
+
+    ReleaseNotes:
+      description: Package release notes.
+      type: string
+      nullable: true
+      minLength: 3
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:736]: >3 [](http://example.com/codeflow?start=17&length=1)

1

File: documentation/WinGet-1.0.0.yaml [hunk 11]
```diff
@@ -1031,10 +1214,24 @@ components:
...
+          $ref: '#/components/schemas/AutoPin'
+        UnsupportedOSArchitectures:
+          $ref: '#/components/schemas/UnsupportedOSArchitectures'
+        AppsAndFeaturesEntries:
+          $ref: '#/components/schemas/AppsAndFeaturesEntries'
+        Markets:
+          $ref: '#/components/schemas/Markets'
       required:
         - Architecture
-        - InstallerUrl
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1036]: >- InstallerUrl [](http://example.com/codeflow?start=8&length=14)

update related description

File: documentation/WinGet-1.0.0.yaml [hunk 12]
```diff
@@ -956,32 +1016,153 @@ components:
...
+
+    Architecture:
+      description: The installer target architecture.
+      oneOf:
+        - $ref: '#/components/schemas/ArchitectureNonNeutral'
+        - type: string
+          enum:
+            - neutral
+
+    ProductId:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1036]: >ProductId [](http://example.com/codeflow?start=4&length=9)

MSStoreProductIdentifier

File: documentation/WinGet-1.0.0.yaml [hunk 13]
```diff
@@ -956,32 +1016,153 @@ components:
...
+      description: The installer target market.
+      type: string
+      nullable: true
+      pattern: "^[A-Z]{2}$"
+
+    Markets:
+      description: The installer markets
+      type: object
+      properties:
+        AllowedMarkets:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1052]: >AllowedMarkets [](http://example.com/codeflow?start=8&length=14)

oneof:

File: documentation/WinGet-1.0.0.yaml [hunk 14]
```diff
@@ -956,32 +1016,153 @@ components:
...
+      description: The installer release date.
+      nullable: true
+      type: string
+      format: date
+      
+    InstallLocationRequired:
+      description: Indicates whether the installer requires an install location provided.
+      type: boolean
+
+    AutoPin:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1081]: > AutoPin [](http://example.com/codeflow?start=3&length=8)

RequiresExplicitUpgrade

File: documentation/WinGet-1.0.0.yaml [hunk 15]
```diff
@@ -956,32 +1016,153 @@ components:
...
+          description: The DisplayName
+          type: string
+          nullable: true
+          minLength: 2
+          maxLength: 256
+        AppPublisher:
+          description: The Publisher
+          type: string
+          nullable: true
+          minLength: 2
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1106]: >2 [](http://example.com/codeflow?start=21&length=1)

1

File: documentation/WinGet-1.0.0.yaml [hunk 16]
```diff
@@ -956,32 +1016,153 @@ components:
...
+        
+    AppsAndFeaturesEntry:
+      description: Various key values under installer's ARP entry.
+      type: object
+      properties:
+        AppName:
+          description: The DisplayName
+          type: string
+          nullable: true
+          minLength: 2
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1100]: >2 [](http://example.com/codeflow?start=21&length=1)

1

File: documentation/WinGet-1.0.0.yaml [hunk 17]
```diff
@@ -956,32 +1016,153 @@ components:
...
     SourceIdentifier:
       description: The source identifier is a unique identifier for the source server.
       type: string
       minLength: 3
       maxLength: 128
+      
+    SourceAgreement:
+      description: The source agreement users must accept before using the source.
+      type: string
+      minLength: 3
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1136]: >3 [](http://example.com/codeflow?start=17&length=1)

1

File: documentation/WinGet-1.0.0.yaml [hunk 18]
```diff
@@ -1003,6 +1184,8 @@ components:
           $ref: '#/components/schemas/MinimumOSVersion'
         InstallerType:
           $ref: '#/components/schemas/InstallerType'
+        NestedInstallerType:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1187]: >NestedInstallerType: [](http://example.com/codeflow?start=8&length=20)

remove

File: documentation/WinGet-1.0.0.yaml [hunk 19]
```diff
@@ -956,32 +1016,153 @@ components:
...
+      description: List of OS architectures the installer does not support.
+      type: array
+      uniqueItems: true
+      items:
+        $ref: '#/components/schemas/ArchitectureNonNeutral'
+        
+    AppsAndFeaturesEntry:
+      description: Various key values under installer's ARP entry.
+      type: object
+      properties:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1095]: >properties: [](http://example.com/codeflow?start=6&length=11)

AlternativeInstallerType:

File: documentation/WinGet-1.0.0.yaml [hunk 20]
```diff
@@ -956,32 +1016,153 @@ components:
...
+        AppName:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1096]: >AppName [](http://example.com/codeflow?start=8&length=7)

update to registry keys

File: documentation/WinGet-1.0.0.yaml [hunk 21]
```diff
@@ -956,32 +1016,153 @@ components:
...
+      items:
+        $ref: '#/components/schemas/AppsAndFeaturesEntry'
 
     SourceIdentifier:
       description: The source identifier is a unique identifier for the source server.
       type: string
       minLength: 3
       maxLength: 128
+      
+    SourceAgreement:
```
Comment (yao-msft) [inline documentation/WinGet-1.0.0.yaml:1133]: >SourceAgreement [](http://example.com/codeflow?start=4&length=15)

SourceAgreements:
$ref: Agreements

**Pull request closed without merging**